integration_tests/src/lib.rs

```diff
@@ -2422,6 +2426,111 @@ mod test {
             )
             .await
             .unwrap_err();
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn integration_test_volume_replace_downstairs_then_takeover(
+    ) -> Result<()> {
+        // Replace a downstairs with a new one, then an Upstairs with a newer
+        // generation number activates.
+        const BLOCK_SIZE: usize = 512;
+
+        // boot three downstairs, write some data to them
+        let test_downstairs_set = TestDownstairsSet::big(false).await?;
+
+        let mut volume = Volume::new(BLOCK_SIZE as u64);
+        volume
+            .add_subvolume_create_guest(
+                test_downstairs_set.opts(),
+                volume::RegionExtentInfo {
+                    block_size: BLOCK_SIZE as u64,
+                    blocks_per_extent: test_downstairs_set.blocks_per_extent(),
+                    extent_count: test_downstairs_set.extent_count(),
+                },
+                1,
+                None,
+            )
+            .await?;
+
+        volume.activate().await?;
+
+        let random_buffer = {
+            let mut random_buffer =
+                vec![0u8; volume.total_size().await? as usize];
+            rand::thread_rng().fill(&mut random_buffer[..]);
+            random_buffer
+        };
+
+        volume
+            .write(
+                Block::new(0, BLOCK_SIZE.trailing_zeros()),
+                Bytes::from(random_buffer.clone()),
+            )
+            .await?;
+
+        // Create a new downstairs, then replace one of our current
+        // downstairs with that new one.
+        let new_downstairs = test_downstairs_set.new_downstairs().await?;
+
+        let res = volume
+            .replace_downstairs(
+                test_downstairs_set.opts().id,
+                test_downstairs_set.downstairs1_address().await,
+                new_downstairs.address().await,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(res, ReplaceResult::Started);
+
+        eprintln!("Wait for some repair work to proceed");
+        tokio::time::sleep(tokio::time::Duration::from_secs(4)).await;
```

Comment:
I've been pulling out hair tracking down failing CI tests, and the sleep here makes me
nervous.  Is there any way to make this deterministic?  Like checking for some job progress but
not completed repair yet?  On a busy system or a fast system, could this sleep be too little
or too much?

Reply:
Good point, I changed this to check for the `StartedAlready` and panic if the repair completed in 9994e79